docs/user_manual/dataset-terminology.md

```diff
@@ -10,23 +10,80 @@ Some terms regarding the data structures are explained here, including the defin
 
 ## Data structures
 
-- **Dataset:** Either a single or a batch dataset.
-    - **SingleDataset:** A data type storing input data (i.e. all elements of all components) for a single scenario.
-    - **BatchDataset:** A data type storing update and or output data for one or more scenarios. A batch dataset can contain sparse or dense data, depending on the component.
-- **DataArray** A data array can be a single or a batch array. It is a numpy structured array.     
-    - **SingleArray** A dictionary where the keys are the component types and the values are one-dimensional structured numpy arrays.
-    - **BatchArray:** An array of dictionaries where the keys are the component types and the values are two-dimensional structured numpy arrays.
-        - **DenseBatchArray:** A two-dimensional structured numpy array containing a list of components of the same type for each scenario.
-        - **SparseBatchArray:** A dictionary with a one-dimensional numpy int64 array and a one-dimensional structured numpy arrays. 
-
-### Type of Dataset 
+```{mermaid}
+graph TD
+    subgraph Datasets
+    Dataset --> SingleDataset
+    Dataset --> BatchDataset
+    end
+
+    subgraph Dataset values
+    ComponentData --> DataArray
+    ComponentData --> ColumnarData
+
+    DataArray --> SingleArray
+    DataArray --> BatchArray
+
+    BatchArray --> DenseBatchArray
+    BatchArray --> SparseBatchArray
+
+    ColumnarData --> SingleColumnarData
+    ColumnarData --> BatchColumnarData
+
+    BatchColumnarData --> DenseColumnarData
+    BatchColumnarData --> SparseColumnarData
+    end
+
+    subgraph Other numpy arrays
+    Indptr
+    SingleColumn
+    BatchColumn
+    end
+```
+
+- **Dataset:** Either a single or a batch dataset. it is a dictionary with keys as the component types (eg. `line`, `node`, etc) and values as **ComponnentData**
```

Comment:
Here and below: should we also refer to the API reference or not cause this is more generic than just Python

Reply:
Good idea. API reference contains the example as well. I was wondering before if i should add them.